### PR TITLE
Display summary columns on external forms review table

### PR DESCRIPTION
**Pull request by gigxz**
## Description

Issue: https://github.com/open-path/Green-River/issues/6622
Depends on hmis-warehouse PR: https://github.com/greenriver/hmis-warehouse/pull/4720

1. Add summary field columns
2. Add link to enrollment on the modal (it felt too busy to add it to the table, and I thought this might be sufficient? but want feedback on that!)

Screenshot 2024-09-18 at 3 50 44 PM

Screenshot 2024-09-18 at 3 50 52 PM

## Type of change
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/modules/projects/components/tables/ProjectExternalSubmissionsTable.tsx [hunk 1]
```diff
@@ -38,43 +42,66 @@ const ProjectExternalSubmissionsTable = ({
...
+            header: key,
+            render: (submission: ExternalFormSubmissionSummaryFragment) =>
+              submission.summaryFields.find((f) => f.key === key)?.value,
+          }));
+
+      return [
+        {
+          header: 'Status',
+          linkTreatment: false,
+          render: (s: ExternalFormSubmissionSummaryFragment) => {
```
Comment (ttoomey) [inline src/modules/projects/components/tables/ProjectExternalSubmissionsTable.tsx]: Maybe avoid single-char vars when referenced on multiple lines? Could replace with destructuring?
```suggestion
          render: ({status, spam, submittedAt}: ExternalFormSubmissionSummaryFragment) => {
```

File: src/modules/projects/components/tables/ProjectExternalSubmissionsTable.tsx [hunk 2]
```diff
@@ -122,15 +149,38 @@ const ProjectExternalSubmissionsTable = ({
...
+          {selected && (
+            <DynamicView
+              values={submissionValues}
+              definition={selected.definition.definition}
+            />
+          )}
+        </Card>
+        {selected?.clientId && selected?.enrollmentId && (
+          <Stack sx={{ mt: 2 }}>
+            <Alert severity='success'>
```
Comment (ttoomey) [inline src/modules/projects/components/tables/ProjectExternalSubmissionsTable.tsx]: should the alert be above the summary (dynamic view)? This is fine but I guess I'm more used to alerts being at the top of the page/modal

**Review by ttoomey — APPROVED** [review]
looks good!

File: src/modules/projects/components/tables/ProjectExternalSubmissionsTable.tsx (hunk 2 above)
Comment (gigxz) [reply src/modules/projects/components/tables/ProjectExternalSubmissionsTable.tsx]: Martha is reworking this dialog in a separate PR, so I was going to let her handle moving this around appropriately!

Comment (martha) [reply src/modules/projects/components/tables/ProjectExternalSubmissionsTable.tsx]: I could go either way, I guess I have an inclination to put it in the same place as the "Mark reviewed" checkbox would go on an as-yet-unreviewed form. Maybe we can check with design about it?

**Review by martha — APPROVED** [review]
Looks great to me. I'm sorry about the merge conflicts! :| Let me know if you would like to pair on resolving them.

**Comment by gigxz** [comment]
No worries it wasn't too bad!

**Pull request merged**